### PR TITLE
Created new customize 404 page. #192

### PR DESCRIPTION
**Pull request by Vivek-04022001**
### Changes that I done: 
1. Created a new **NotFoundPage component** in the components folder.
2. Added a **404.tsx file** and used the **NotFoundPage component** in it.
3. Implemented a check in **_app.tsx** using **useRouter** to determine if the current page is not the home page. If it's not, the new **NotFoundPage** is displayed.
4. Made a **modification to tailwind.config.js** by adding a new CSS class, **text-5xl, and set its value to 7rem.**

### CODE REVIEW

File: components/NotFoundPage.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import Link from "next/link";
+
+const NotFoundPage = () => {
```
Comment (lucavallin) [inline components/NotFoundPage.tsx:3]: Since this is not a page, I would call it `NotFound`

File: components/NotFoundPage.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  return (
+    <div className="flex min-h-screen flex-col items-center  justify-center space-y-10 bg-dark-400 text-light-300 antialiased ">
+      <h3 className="text-3xl font-bold text-primary lg:text-5xl">404😅</h3>
+
+      <h3 className="text-xl lg:text-3xl">Look Like you&apos;re lost</h3>
```
Comment (lucavallin) [inline components/NotFoundPage.tsx:8]:
```suggestion
      <h3 className="text-xl lg:text-3xl">Whooops! We could not find that.</h3>
```

File: components/NotFoundPage.tsx [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+      <p className="text-secondary">the page you are looking for not available</p>
```
Comment (lucavallin) [inline components/NotFoundPage.tsx:9]:
```suggestion
      <p className="text-secondary">the page you are looking for does not exist</p>
```

File: components/NotFoundPage.tsx [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+      <Link href="/" className="text-white border border-primary p-4 hover:bg-primary">
+        Go back to home
```
Comment (lucavallin) [inline components/NotFoundPage.tsx:11]:
```suggestion
        Go back
```

File: pages/_app.tsx [hunk 5]
```diff
@@ -19,17 +21,25 @@ const inter = Inter({
 
 // Entry point for the app
 export default function App({ Component, pageProps }: AppProps) {
+  const router = useRouter();
+  const shouldRenderComponents = router.pathname === "/";
```
Comment (lucavallin) [inline pages/_app.tsx:25]: This one breaks the site. Legit URLs like `/language/html` return 404.

**Comment by lucavallin** [comment]
Thanks for the PR! Left a few comments.

File: pages/_app.tsx (hunk 5 above)
Comment (Vivek-04022001) [reply pages/_app.tsx:25]: I've identified the error, and I've also added the URLs for the tags:
```
const shouldRenderComponents = router.pathname === "/" || router.pathname.startsWith('/language/') || router.pathname.startsWith('/tag/');
```

File: components/NotFoundPage.tsx (hunk 4 above)
Comment (Vivek-04022001) [reply components/NotFoundPage.tsx:11]: I've reviewed all the suggestions and have adjusted my code accordingly
image

**Comment by Vivek-04022001** [comment]
@lucavallin, I want to express my gratitude for your valuable suggestions. I apologize for any inconvenience caused. This marks my first pull request, and I've incorporated the changes in my second commit. I'm eager to continue contributing to this fantastic project. Please feel free to provide further guidance

**Comment by lucavallin** [comment]
@Vivek-04022001 Thanks, almost perfect. The "Go back" button should follow the style of other buttons, for example:

image

- Uppercase
- Rounded
- Bold/whatever font we have elsewhere

**Comment by Vivek-04022001** [comment]
okay, I will create a new pull request. Thanks for the suggestions. Im so excited, i love contribute to your project

**Comment by lucavallin** [comment]
@Vivek-04022001 Looks good, please push your changes.

**Review by lucavallin — APPROVED** [review]
Good work!

**Pull request merged**